lib/active_force/association/has_many_association.rb

```diff
@@ -13,13 +25,7 @@ def define_relation_method
         @parent.send :define_method, _method do
           association_cache.fetch _method do
             query = association.relation_model.query
-            if scope = association.options[:scoped_as]
-              if scope.arity > 0
-                query.instance_exec self, &scope
-              else
-                query.instance_exec &scope
-              end
-            end
+            apply_scope query
```

Comment:
This is within a `define_method` block, which is adding to the API of `@parent`. I don't think you will have access to `apply_scope` when the dynamically defined method is invoked.